security/token-file-provider.1.rst

```diff
@@ -0,0 +1,175 @@
+NAME
+====
+token-file-provider – Generate Vault tokens for EdgeX services
+
+
+SYNOPSIS
+========
+token-file-provider [-h|-help] [-c|--confdir <confdir>] [-p|--profile <name>]
+
+
+DESCRIPTION
+===========
+token-file-provider generates per-service Vault tokens for EdgeX services
+so that they can make authenticated connections to Vault to retrieve
+application secrets.
+token-file-provider implements a generic secret seeding mechanism based
+on pre-created files and is designed for maximum portability.
+token-file-provider takes a configuration file that specifies the services
+for which tokens shall be generated and the Vault access policy
+that shall be applied to those tokens.
+token-file-provider assumes that there is some underlying protection mechanism
+that will be used to prevent EdgeX services from reading each other’s tokens.
+
+
+OPTIONS
+=======
+  \-h, \--help
+    Display help text
+
+  \-c, \--confdir <confdir>
+    Look in this directory for configuration.toml instead.
+
+  \-p, \--profile <name>
+    Indicate configuration profile other than default
+
+
+FILES
+=====
+
+configuration.toml
+------------------
+This file specifies the TCP/IP location of the Vault service
+and parameters used for Vault token generation.
+
+::
+
+  [SecretService]
+  Server = "localhost"
+  Port = 8200 
+
+  [TokenFileProvider]
+  PrivilegedTokenPath = /run/edgex/secrets/token-file-provider/vault-token.json
+  ConfigFile = token-config.json
+  OutputDir = /run/edgex/secrets/
+  OutputFilename = vault-token.json
```

Comment:
the PrivilegedTokenPath contains the OutputFilename, and OutputDir
guess "token-file-provider" is the specific implementation.
Just curious if we need OutputDir and OutputFilename.

Reply:
We can hard-code `OutputFilename` if you wish.

Reply:
Can't we just have one key with the directory and the filename?

Reply:
Each service gets their own token -- you mean like `/dir/%s/vault-token.json` ?

Reply:
ah I missed the actual purpose of this, I suppose that's intended to happen when just reviewing docs for code that hasn't been written yet.
So really what `OutputDir` and `OutputFilename` does is generate a list of files matching the pattern `$OutputDir/$ServiceName/$OutputFileName`. I think having these separate config options is fine, but when this is implemented in code, let's revisit what to call these options because as written it's not clear how these things interact.